### ISSUE TITLE
Test suite fails a lot due to time zone weirdness maybe

### ISSUE THREAD

**Issue by awilfox** [oetiker/rrdtool-1.x#896]
Hi there.

I'm trying to run the RRDtool 1.7.0 test suite for packaging on Adélie Linux.  8 of the 13 tests fail, and 7 of them fail due to what seems like time zone weirdness.  Possibly tzdata wrongly changed the way CET works (it should not have affected dates in the past, but it looks like it did):

```diff
--- -   2018-07-25 10:23:04.098819439 +0000
+++ /usr/src/packages/user/rrdtool/src/rrdtool-1.7.0/tests/modify-test1.create.dump     2015-11-10 15:50:57.000000000 +0000
@@ -4,7 +4,7 @@
 <rrd>
        <version>0003</version>
        <step>60</step> <!-- Seconds -->
-       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 07:57:00 CET -->
+       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 08:57:00 CET -->

        <ds>
                <name> a </name>
```

```diff
--- /usr/src/packages/user/rrdtool/src/rrdtool-1.7.0/tests/modify2-testa-create.dump    2015-11-10 15:50:57.000000000 +0000
+++ -   2018-07-25 10:23:04.121496636 +0000
@@ -4,7 +4,7 @@
 <rrd>
        <version>0003</version>
        <step>60</step> <!-- Seconds -->
-       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 08:57:00 CET -->
+       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 07:57:00 CET -->

        <ds>
                <name> a </name>
```

```diff
--- -   2018-07-25 10:23:04.100982894 +0000
+++ /usr/src/packages/user/rrdtool/src/rrdtool-1.7.0/tests/modify-test3.create.dump     2015-11-10 15:50:57.000000000 +0000
@@ -4,7 +4,7 @@
 <rrd>
        <version>0003</version>
        <step>60</step> <!-- Seconds -->
-       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 07:57:00 CET -->
+       <lastupdate>1300003020</lastupdate> <!-- 2011-03-13 08:57:00 CET -->

        <ds>
                <name> a </name>
```

Should the tests be changed, perhaps to use UTC which should (hopefully!) never change?

**Comment by c72578** [comment]
Hello,
do you still observe this behaviour using the current version of Adélie Linux (e.g.  1.0-BETA2)?
What is the output on the command line on your system of:
`TZ='CET' date -d @1300003020`

**Comment by awilfox** [comment]
Yes, I am still observing this behaviour.  We are using GCC 8.3 now, if that matters.

```
awilcox on gwyn [pts/2 Sat 23 22:11] rrdtool: TZ='CET' date -d @1300003020
Sun Mar 13 07:57:00 CET 2011
```

**Comment by c72578** [comment]
OK, thanks for testing.
This means, that TZ='CET' has no effect and the time is shown as UTC.
I just did some tests under Adélie Linux 1.0-beta2 and it seems that deprecated tz database time zones are not working, e.g. CET, EST etc.
List: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
These are examples for working ones:
Etc/GMT-1
Europe/Zurich
Interesting:
`TZ='Europe/Zurich' date -d @1300003020`
`Sun Mar 13 08:57:00 CET 2011`
So, in this case the time is correct and CET is shown in the output.

**Comment by c72578** [comment]
What you could do as a quick workaround for packaging rrdtool under Adélie Linux, where the tests will pass, is patching `tests/functions` and change `TZ` e.g. to:
```diff
-TZ=CET
+TZ='Europe/Zurich'
```

**Comment by c72578** [comment]
Fixed in v1.7.2.
Relevant commit: https://github.com/oetiker/rrdtool-1.x/commit/fc4527e12046dad5f6a0e80398a310c8e5e343d0
